core/src/main/java/me/mykindos/betterpvp/core/combat/weapon/Weapon.java

```diff
@@ -41,6 +56,64 @@ public void loadWeapon(BPVPItem item) {
         setGiveUUID(item.isGiveUUID());
     }
 
+    /**
+     * @param name         name of the value
+     * @param defaultValue default value
+     * @param type         The type of default value
+     * @param <T>          The type of default value
+     * @return returns the config value if exists, or the default value if it does not. Saves the default value if no value exists
+     */
+    protected <T> T getConfig(String name, Object defaultValue, Class<T> type) {
+        return plugin.getConfig(getConfigName()).getOrSaveObject(getPath(name), defaultValue, type);
+    }
+
+    /**
+     * @param name         name of the value
+     * @param defaultValue default value
+     * @param type         The type of default value
+     * @param <T>          The type of default value
+     * @return returns the config value if exists, or the default value if it does not. Does not save value in the config
+     */
+    protected <T> T getConfigObject(String name, T defaultValue, Class<T> type) {
+        return plugin.getConfig(getConfigName()).getObject(getPath(name), type, defaultValue);
+    }
+
+    protected String getConfigName() {
+        String weaponType = "standard";
+        if (this instanceof LegendaryWeapon) {
+            weaponType = "legendaries";
+        }
+        return "weapons/" + weaponType;
+    }
```

Comment:
I'd prefer a default implementation in IWeapon which return "weapons/standard", then another default implementation in the LegendaryWeapon interface which would be weapons/legendaries. This will give us the ability to override it easily without making this function longer and longer

Reply:
Makes sense, implemented